overlord/configstate/configcore/corecfg.go

```diff
@@ -62,6 +63,83 @@ func validateBoolFlag(tr config.Conf, flag string) error {
 	return nil
 }
 
+// PlainCoreConfig carries a read-only copy of core config and implements
+// config.ConfReader interface.
+type PlainCoreConfig map[string]interface{}
+
+// Get implements config.ConfReader interface.
+func (cfg PlainCoreConfig) Get(snapName, key string, result interface{}) error {
+	if snapName != "core" {
+		return fmt.Errorf("internal error: expected core snap in Get(), %q was requested", snapName)
+	}
+
+	// XXX: should we support nesting with dotted key?
```

Comment:
not particularly,  but we need to add a function or method in the gadget package that reads out system/core config from gadget.yaml defaults in a format that can be consumed here?

Reply:
Right, most likely yes. Would you like this here or in a followup (either way works for me)?

Reply:
follow up is fine, but we do need it soon to use this I think.

Reply:
we should drop the XXX then